### ISSUE TITLE
Listeners are not stopped when Junit5-Test-Extension shuts down Spring

### ISSUE THREAD

**Issue by guenhter** [spring-projects/spring-amqp#741]
Hi,

When the JUnit5-Extension if used with Spring, it seems that registered `RabbitListener` are not shutdown, even if all the `after...` methods of the extension are called.
So when running a JUnit5-Test with Spring, and then stopping in another Test (e.g. Breakpoint), the connections to Rabbit from the first test are still open.

I expect, that on shutting down the Test, also the Listeners goes down.

cheers

**Comment by garyrussell** [comment]
Right, the application context is cached, in case you want to use the same context in other test cases, adding `@DirtiesContext` at the class level will shut down the context after all the tests in the class have completed.

**Comment by guenhter** [comment]
What I actually ment is, that the Listener lives longer than the Test-Class.

I reproduces the behavior with 
https://github.com/guenhter/spring-rabbit-test-closing-bug

Don't forget to start a rabbitmq instance before running the tests.

There, two tests are contained: `Test1`, `Test2`. Both do exaclty the same: 
* sending 1000 messages to rabbit
* receiving those messages and assert that
** it are 1000
** the prefix is correct (test dependent prefix)

Running the tests individually works fine, but running both tests in one run, the second one fails, because there is still a Listener from the first Test-Class active.

**Comment by garyrussell** [comment]
It's also good practice to use different queues for each test, to further isolate them.

Also, consider using the `@RabbitAvailable` annotation with the queue - it uses the JUnit5 `RabbitAvailableCondition` - which will ensure the queue is empty before the test starts (and create it if necessary) and will delete it when the tests complete.

**Comment by guenhter** [comment]
Thanks. And yes, I'll use Stackoverflow in the future at first.
